flux/README.md

```diff
@@ -0,0 +1,262 @@
+   booktitle={The Thirteenth International Conference on Learning Representations},
+   year={2025},
+   url={https://openreview.net/forum?id=SFN6Wm7YBI}
+}
+```
+
+# 2. Directions
+### Steps to configure machine
+To use this repository, please ensure your system can run docker containers and has appropriate GPU support (e.g. for CUDA GPUs, please make sure the appropriate drivers are set up)
+
+Without docker, follow the [instructions](https://github.com/pytorch/torchtitan?tab=readme-ov-file#installation) to install torchtitan and additionally install `requirements-mlperf.txt` and `torchtitan/experiments/flux/requirements.txt`.
+
+### Container setup
+To build the container:
+```bash
+docker build -t <tag> -f Dockerfile .
+```
+
+Before entering the container, create a directory for the models to be downloaded, and a directory to be used as huggingface cache (necessary for some operations):
+
+```bash
+mkdir models
+mkdir hf_cache
+```
+
+```
+docker run -it --rm \
+--gpus all --ulimit memlock=-1 --ulimit stack=67108864 \
+--network=host --ipc=host \
+-v ~/.ssh:/root/.ssh \
+-v hf_cache:/root/.cache \
+-v <path for dataset storage>:/dataset \
+-v <path for model storage>:/models \
+<tag> bash
+```
+Note: it's recommended to map your .ssh folder to inside the container, so that it's easier for the code to set up remote cluster access.
+
+### Steps to download and verify data
+For all steps below, they are assumed to run inside the container
+
+#### CC12M dataset
+Download the dataset with
+```bash
+HF_TRANSFER=1 huggingface-cli download --repo-type dataset pixparse/cc12m-wds --local-dir /dataset/cc12m-wds
+```
+Then, we remove problematic indices and keep only the first 10% of this data (rounded to 1,099,776 samples so it is nicely divisible by large powers of 2).
+Depending on your CPU, you may wish to change `--num_workers` and `--batch_size`. This only impacts the runtime of this script,
+the final result will be not be affected by these parameters.
+
+```bash
+python torchtitan/experiments/flux/scripts/clean_cc12m.py --input_dir /dataset/cc12m-wds --output_dir /dataset/cc12m_disk --filter_file torchtitan/experiments/flux/scripts/problematic_indices.txt --num_workers=16 --batch_size 1000
+```
+(Optional) Remove the downloaded dataset to reclaim space: `rm -r /dataset/cc12m-wds`
+
+The filter file is included in this repository. It was generated using `torchtitan/experiments/flux/scripts/find_problematic_indices.py`.
+
+#### COCO-2014 subset
+The number of samples is taken from the previous stable diffusion benchmark, but rounded slightly to be divisible by large powers of 2.
+
+1. download coco-2014 validation dataset: `DOWNLOAD_PATH=/dataset/coco2014_raw bash torchtitan/experiments/flux/scripts/coco-2014-validation-download.sh`
```

Comment:
why are these four separate scripts? Is there a need to have intermediate results?